src/ML/ML.hpp

```diff
@@ -122,9 +138,13 @@ void NeuralNetworkComponent<TNeuralNetwork>::MakeDecision()
 
     // Collect new input
     collectInput(decisionInput.data.get()[TNeuralNetwork::SequenceLength - 1]);
+	inputsCollected++;
```

Comment:
Generally a the pre-increment operator is a touch better to use (++inputsCollected). But ultimately it doesn't matter on a line on its own, more so a stylistic decision... (I believe most modern compilers understand to covert post-increment to pre-increment operations if it is doable)

Still though the rest of the code uses pre increments, so I'd suggest using those instead.